src/games/strategy/ui/SwingWorkerCompletionWaiter.java

```diff
@@ -0,0 +1,82 @@
+package games.strategy.ui;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.awt.Window;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+
+import javax.swing.SwingWorker;
+
+/**
+ * Manages the display of a window during the execution of a {@code SwingWorker}.
+ *
+ * <p>
+ * Use this class when it is necessary to display progress during a potentially long-running operation. If the <i>Event
+ * Dispatch Thread</i> must be blocked until the operation is complete (i.e. the user should not be allowed to perform
+ * any other actions in the UI), it is recommended you use a modal dialog.
+ * </p>
+ *
+ * <p>
+ * For example:
+ * </p>
+ *
+ * <pre>
+ * JDialog dialog = new JDialog(owner, true);
+ * swingWorker.addPropertyChangeListener(
+ *     new SwingWorkerCompletionWaiter(dialog));
+ * swingWorker.execute();
+ * // the dialog will be visible until the SwingWorker is done
+ * dialog.setVisible(true);
+ * </pre>
+ */
+public final class SwingWorkerCompletionWaiter implements PropertyChangeListener {
+  static final String SWING_WORKER_STATE_PROPERTY_NAME = "state";
+
+  private final IWindow window;
+
+  /**
+   * Initializes a new instance of the {@code SwingWorkerCompletionWaiter} class.
+   *
+   * @param window The window to display while the operation is in progress; must not be {@code null}.
+   */
+  public SwingWorkerCompletionWaiter(final Window window) {
+    this.window = IWindow.fromAwt(checkNotNull(window));
+  }
+
+  SwingWorkerCompletionWaiter(final IWindow window) {
+    this.window = window;
+  }
+
+  @Override
+  public void propertyChange(final PropertyChangeEvent event) {
+    if (SWING_WORKER_STATE_PROPERTY_NAME.equals(event.getPropertyName())) {
+      if (SwingWorker.StateValue.STARTED == event.getNewValue()) {
+        window.setVisible(true);
+      } else if (SwingWorker.StateValue.DONE == event.getNewValue()) {
+        window.setVisible(false);
+        window.dispose();
+      }
+    }
+  }
+
+  static interface IWindow {
```

Comment:
Please avoid/remove the `I` prefix on interfaces. (http://www.triplea-game.org/dev_docs/dev/code_format - http://google.github.io/styleguide/javaguide.html#s5.1-identifier-names)

Doing stuff specifically for test only, can be a smell, yes. In this case you are allowing hard to test code to be testable. You lose perhaps one point for the test scaffolding, but gain perhaps about 5 for testable code. In another sense you are also decoupling the UI interaction API with the UI implementation. That is a very good thing and is worth perhaps 5 more "points."  :+1: 

Looking at the API and usage, there is not strict need to keep the exact Swing API. You could combine both the `dispose` and `setVisible` methods into perhaps a `close()` method, where you would call both.

The static method to do a default conversion I think is pretty clean. Overall this type of interface is probably a good pattern for a first iteration of getting some of the Swing code out of the game logic code.  So overall :+1:  from me here.